tests/rbd_mirror/test_rbd_mirror_secondary_full.py

```diff
@@ -0,0 +1,134 @@
+    when secondary cluster storage is full.
+
+    Args:
+        **kw: test data
+
+    Returns:
+        0 - if test case pass
+        1 - if test case fails
+    """
+
+    try:
+        log.info(
+            "Starting CEPH-9507 - Verification of mirroring failure "
+            "when secondary cluster storage is full"
+        )
+        rbd = Rbd(**kw)
+        config = kw.get("config")
+        mode = config.get("mode")
+        mirrormode = config.get("mirrormode")
+        poolname = rbd.random_string() + "_test_pool"
+        imagename = rbd.random_string() + "_test_image"
+
+        rbd1, rbd2 = [
+            Rbd(**kw, req_cname=cluster_name)
+            for cluster_name in kw.get("ceph_cluster_dict").keys()
+        ]
+
+        mirror1 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd1"), config
+        )
+        mirror2 = rbdmirror.RbdMirror(
+            kw.get("ceph_cluster_dict").get("ceph-rbd2"), config
+        )
+
+        cluster_stats = get_storage_stats(rbd1.ceph_client)
+        cluster1_size = convert_bytes(cluster_stats.get("total_avail_bytes"), "gb")
+
+        cluster_stats = get_storage_stats(rbd2.ceph_client)
+        cluster2_size = convert_bytes(cluster_stats.get("total_avail_bytes"), "gb")
+
+        if cluster2_size > cluster1_size:
+            log.error(
+                "keep secondary storage space less compare to primary to perform this test"
+            )
+            return 1
+
+        image_size = str(cluster2_size) + "GB"
+
+        # Calculate 80% of the image size
+        io_size = int(cluster2_size * 0.8)
+
+        mirror1.initial_mirror_config(
+            mirror2,
+            poolname=poolname,
+            imagename=imagename,
+            imagesize=image_size,
+            io_total=config.get("io-total", "10G"),
+            mode=mode,
+            mirrormode=mirrormode,
+            peer_mode=config.get("peer_mode", "bootstrap"),
+            rbd_client=config.get("rbd_client", "client.admin"),
+            build=config.get("build"),
+            **kw,
+        )
+
+        mirror1.mirror_snapshot_schedule_add(poolname=poolname, interval="3m")
+
+        size = int(io_size / 10)
+
+        try:
+            with parallel() as p:
+                for _ in range(10):
+                    p.spawn(
+                        run_fio,
+                        image_name=imagename,
+                        pool_name=poolname,
+                        size=str(size) + "GB",
+                        timeout="notimeout",
+                        client_node=rbd1.ceph_client,
+                    )
+        except Exception as e:
+            log.info(
+                f"An error occurred: {e}. Expected due to secondary cluster full IO gets stop"
+            )
+
+        # Validation for secondary full from secondary
+        out = mirror2.exec_cmd(cmd="ceph -s", output=True)
+
+        if "HEALTH_ERR" or "HEALTH_WARN" in out:
```

Comment:
I think we discussed that the validation here should be that, it should be a HEALTH_ERR and also the secondary cluster health should say atleast one of the OSDs is full.

Please implement validation steps accordingly.

Reply:
Done added